packages/xeus-extension/src/index.ts

```diff
@@ -2,133 +2,192 @@
+      let kernelId: string | undefined;
+      let kernelName: string | undefined;
+      let logConsolePanel: LogConsolePanel | undefined;
+      let sourceId: string | undefined;
+      let logConsoleWidget: MainAreaWidget<LogConsolePanel> | undefined;
+
+      const createWidget = () => {
+        if (!logConsolePanel) {
+          return;
+        }
+
+        logConsoleWidget = new MainAreaWidget<LogConsolePanel>({
+          content: logConsolePanel
+        });
+        logConsoleWidget.title.label = 'Kernel Logs';
+        logConsoleWidget.title.icon = listIcon;
+        logConsoleWidget.id = `${sourceId}-widget`;
+      };
+
+      const toolbarButton = new ToolbarButton({
+        icon: listIcon,
+        onClick: () => {
+          if (!logConsolePanel) {
+            showErrorMessage(
+              'Cannot show logs',
+              'Cannot show logs for the current kernel'
             );
+            return;
           }
-          const link = empackEnvMetaFile
-            ? await empackEnvMetaFile.getLink(kernelspec)
-            : '';
-
-          return new WebWorkerKernel({
-            ...options,
-            contentsManager,
-            mountDrive,
-            kernelSpec: kernelspec,
-            empackEnvMetaLink: link
-          });
-        }
+
+          if (!logConsoleWidget) {
+            createWidget();
+          }
+
+          if (logConsoleWidget) {
+            if (logConsoleWidget.isDisposed) {
+              createWidget();
+            }
+
+            if (!logConsoleWidget.isAttached) {
+              app.shell.add(logConsoleWidget, 'main', { mode: 'split-bottom' });
+            }
+
+            app.shell.activateById(logConsoleWidget.id);
+          }
+        },
+        tooltip: 'Show kernel logs'
       });
-    }
-    await app.serviceManager.kernelspecs.refreshSpecs();
-  }
-};
 
-const empackEnvMetaPlugin: JupyterLiteServerPlugin<IEmpackEnvMetaFile> = {
-  id: '@jupyterlite/xeus:empack-env-meta',
-  autoStart: true,
-  provides: IEmpackEnvMetaFile,
-  activate: (): IEmpackEnvMetaFile => {
-    return {
-      getLink: async (kernelspec: Record<string, any>) => {
-        const kernelName = kernelspec.name;
-        const kernel_root_url = URLExt.join(
-          PageConfig.getBaseUrl(),
-          `xeus/kernels/${kernelName}`
+      session.kernelChanged.connect(() => {
+        kernelId = session.session?.id;
+        kernelName = session.session?.kernel?.name;
+
+        currentState = KernelStatus.None;
+
+        if (!kernelId || !kernelName) {
+          return;
+        }
+
+        sourceId = `${kernelName}-${kernelId}`;
+
+        logConsolePanel = new LogConsolePanel(
+          new LoggerRegistry({
+            defaultRendermime: rendermime,
+            maxLength: 1000
+          })
         );
-        return `${kernel_root_url}`;
-      }
+
+        logConsolePanel.source = sourceId;
+
+        const channel = new BroadcastChannel(`/kernel-broadcast/${kernelId}`);
```

Comment:
What was the motivation for choosing a `BroadcastChannel` instead of a `MessageChannel`?